### ISSUE TITLE
Upcoming Invoice cannot return Invoice Object

### ISSUE THREAD

**Issue by riveramj** [farmdawgnation/streifen#4]
`getUpcoming` returns an invoice w/o an `id`, which when you try to parse the object as an `invoice`, it fails. We need to be able to generate an `invoice` w/o an `id` for this call.

=========
Alright so i've played around w/this enough to be reasonably sure its not my code that's broke.

I can curl the w/a `customerId` and everything is kosher. Likewise, my other stripe calls are working fine.

call in question: https://stripe.com/docs/api#upcoming_invoice & https://github.com/farmdawgnation/streifen/blob/master/src/main/scala/me/frmr/stripe/Invoice.scala#L96

I should be able to give it a valid `customerId` and get back an `invoice` object. 

`Invoice.getUpcoming(customerId)` gives 

```
get upcoming invoice failed with stripe error: Failure(No usable value for id
Did not find value which can be converted into java.lang.String,Full(net.liftweb.json.MappingException: No usable value for id
Did not find value which can be converted into java.lang.String),Empty)
```

I'm gonna keep working on it, but any help would be appreciated.

**Comment by riveramj** [comment]
I can open a PR w/a base `invoice` trait that is extended by a `Invoice` & `UpcomingInvoice` if you want. 

I'm not sure of any other way to solve this problem.

**Comment by riveramj** [comment]
It couldnt be that easy could it.

getting a `Failure(Parsed JSON values do not match with class constructor` 

Only diff in data structure between invoice and upcoming invoice is 

Upcoming:
```
"lines": {
    "object": "list",
    "data": [
```

vs

Normal Invoice:
```
"lines": {
    "data": [
```

https://github.com/farmdawgnation/streifen/pull/5

**Comment by farmdawgnation** [comment]
Grr. Yeah, this is an annoying "feature" of Stripe's API. Good catch. Let me take a look at the PR.

**Comment by farmdawgnation** [comment]
I believe we can call this solved now, yes?

**Comment by riveramj** [comment]
We solved in #6
